ttnn/cpp/ttnn/operations/data_movement/tilize/device/tilize_program_factory.cpp

```diff
@@ -236,19 +236,19 @@ operation::ProgramWithCallbacks tilize_multi_core_interleaved(const Tensor& a, T
         const CoreCoord& core = cores[i];
 
         // reader runtime args
-        vector<uint32_t> reader_rt_args = {
+        const std::array reader_rt_args = {
```

Comment:
is it possible to specify type once here instead of 3 times below?

Reply:
is the problem in length?

Reply:
The problem is that Constructor Template Argument Deduction (CTAD) is all-or-nothing. Yes, I need to count the number of elements and pass that value explicitly as a template parameter, which is error-prone (you can make that template parameter larger than the number of explicit elements, and it's not a compilation error).